fields/types/gcsimage/GcsImageType.js

```diff
@@ -66,405 +66,336 @@ Object.defineProperty(gcsimage.prototype, 'gcsConfig', {
-
-						promises.push(gcsHelper.uploadFileToBucket(bucket, resizeFunc(file.path, resizeOpt.width, resizeOpt.height, resizeOpt.options), {
-							destination: gcsDir + filenameWithoutExt + '-' + resizeOpt.target + '.' + ext,
-							filetype: filetype,
-							isPublicRead: isPublicRead,
-							cacheControl: 'public, max-age=' + ONE_YEAR,
-						}));
-					}
-					return Promise.all(promises).then(function (values) {
-						return targets;
-					});
-				} else {
-						// skip resizing
-					return;
-				}
-			} else {
-					// skip resizing
-				return;
-			}
-		}).then(function (targets) {
-			var dimensions = sizeOf.sync(fs.readFileSync(file.path));
-
-			if (targets) {
-				try {
-					_.forEach(targets, function (target) {
-						if (typeof target.width === 'number' && !target.height) {
-							if (target.width < dimensions.width) {
-								target.height = Math.round((target.width / dimensions.width) * dimensions.height);
-							} else {
-								target.width = dimensions.width;
-								target.height = dimensions.height;
-							}
-						} else if (typeof target.height === 'number' && !target.width) {
-							if (target.height < dimensions.height) {
-								target.width = Math.round((target.height / dimensions.height) * dimensions.width);
-							} else {
-								target.width = dimensions.width;
-								target.height = dimensions.height;
-							}
-						}
-					});
-				} catch (e) {
-					console.warn('Calculating width and height of resized image occurs error ', e);
-				}
-			}
-			var imageData = {
-				filename: filename,
-				filetype: filetype,
-				gcsBucket: _this.options.bucket,
-				gcsDir: gcsDir,
-				height: dimensions.height || 0,
-				iptc: {},
-				resizedTargets: targets || {},
-				size: file.size,
-				url: gcsHelper.getPublicUrl(_this.options.bucket, gcsDir + filename),
-				width: dimensions.width || 0,
-			};
-
-			if (typeof _this.options.extractIPTC === 'function') {
-				return _this.options.extractIPTC(file.path)
-						.then(function (meta) {
-							imageData.iptc = meta;
-							if (update) {
-								item.set(_this.path, imageData);
-							}
-							return callback(null, imageData);
-						});
-			} else {
-				if (update) {
-					item.set(_this.path, imageData);
-				}
-				callback(null, imageData);
-			}
-		}).catch(function (err) {
-			console.error('UPLOADING ERROR:', err);
-			bucket.deleteFiles({
-				prefix: gcsDir + filenameWithoutExt,
-			}, function (deleteErr) {
-				if (deleteErr) {
-					return callback(deleteErr);
-				}
-				callback(err);
-			});
-		}).then(function () {
-			// delete local file
-			console.log('DELETE LOCAL FILE:', file.path);
-			try {
-				fs.unlinkSync(file.path);
-			} catch (err) {
-				console.error('DELETE LOCAL FILE ERROR:', err);
-			};
-		});
-	};
-
-	doUpload();
+  var _this = this;
+  var ONE_YEAR = 60 * 60 * 24 * 365;
+  var gcsDir = this.options.destination ? this.options.destination : '';
```

Comment:
Minor. I suggest consolidating the `options.destination` to `options.gcsDir` for removing the ambiguity between `gcsHelper.options` and `this.options`

Reply:
This is a good suggestion.
However, in order to make this PR simple,
I would like have another PR to refactor it.